### PR TITLE
Add NYU Depth V2 dataset

### PR DESCRIPTION
**Pull request by leonoverweel**
This is a revision of #1259 by @lgeiger (we work together), which was not merged because [it was an unofficial version of the dataset](https://github.com/tensorflow/datasets/pull/1259#issuecomment-562751279).

The NYU Depth V2 dataset is a bit of a special case. The version on [the official webpage](https://cs.nyu.edu/~silberman/datasets/nyu_depth_v2.html) includes several pieces:
* "1449 densely labeled pairs of aligned RGB and depth images"
* (Much more) "raw rgb, depth and accelerometer data as provided by the Kinect," which is not carefully labeled or cleaned up
* A matlab toolbox that includes scripts for filling in missing depth values (`fill_depth_colorization.m` and `fill_depth_cross_bf.m`)

Because the 1449 densely labeled pairs are not enough to train a model on, the literature has standardised on using the raw data, preprocessed using the matlab script, as training data, and (a subset of) the densely labeled pairs as testing data. Examples of this include:
* [DenseDepth](https://github.com/ialhashim/DenseDepth) by [Alhashim and Wonka (2018)](https://arxiv.org/abs/1812.11941); section 4.1 describes that they use the Levin et al. colorization method mentioned on the official dataset page to fill missing values, presumably using `fill_depth_colorization.m` (although this is not explicitly mentioned).
* [sparse-to-dense](https://github.com/fangchangma/sparse-to-dense) by [Ma and Karaman (2017)](https://arxiv.org/abs/1709.07492); section IV.A. describes that they use `fill_depth_cross_bf.m` to fill missing values.
* [fast-depth](https://github.com/dwofk/fast-depth) by [Wofk et al. (2019)](https://arxiv.org/abs/1903.03273); uses the same dataset as sparse-to-dense.

Since the matlab script makes it impossible to use the raw official dataset in an all-Python machine learning pipeline/setup, researchers have run the official dataset through these preprocessing scripts and uploaded the results to file servers so that they can provide runnable Python reproduction code alongside their papers. #1259 was an example of this; it has the version from Alhashim and Wonka (2018).

We've now found another example of this, hosted by the MIT Laboratory for Information & Decision Systems ([LIDS](https://lids.mit.edu/)), available [here](http://datasets.lids.mit.edu/fastdepth/data/). This version is used by both Ma and Karaman (2017) and Wofk et al. (2019). We think we can consider this MIT version official enough to be included in TensorFlow datasets.

Additionally, we've run this version of the dataset through the dataset definition included in this PR locally, and examined the data to ensure it looks as expected. It does, and the rendered Jupyter notebook is available [here](https://gist.github.com/leonoverweel/deb71bd15ec9170e66b51abac8281875). As you can see, the test data (coming from densely labeled pairs) is quite clean, while the training data is a bit more messy and includes unlabeled regions (zeros).

cc @Conchylicultor, @cyfra

### CODE REVIEW

**Comment by leonoverweel** [comment]
@googlebot I signed it!

**Comment by lgeiger** [comment]
@googlebot I consent.

**Comment by leonoverweel** [comment]
Hey @Conchylicultor @cyfra, any updates/feedback on this? Thanks :)

File: tensorflow_datasets/image/nyu_depth_v2.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+"""NYU Depth V2 Dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import h5py
+import os
+import numpy as np
+import tensorflow as tf
```
Comment (ChanchalKumarMaji) [inline tensorflow_datasets/image/nyu_depth_v2.py]: change to `import tensorflow.compat.v2 as tf`

File: tensorflow_datasets/image/nyu_depth_v2.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+"""
+
+
+_URL = 'http://datasets.lids.mit.edu/fastdepth/data/nyudepthv2.tar.gz'
+
+
+class NyuDepthV2(tfds.core.GeneratorBasedBuilder):
+  """NYU Depth V2 Dataset."""
+
+  VERSION = tfds.core.Version('2.0.0')
```
Comment (ChanchalKumarMaji) [inline tensorflow_datasets/image/nyu_depth_v2.py]: change version to `0.0.1`.

File: tensorflow_datasets/image/nyu_depth_v2.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            gen_kwargs={
+                'root_dir': os.path.join(base_path, 'nyudepthv2', 'train')
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
```
Comment (ChanchalKumarMaji) [inline tensorflow_datasets/image/nyu_depth_v2.py]: `name=tfds.Split.VALIDATION`
https://www.tensorflow.org/datasets/api_docs/python/tfds/Split

**Comment by leonoverweel** [comment]
Hi @ChanchalKumarMaji, thanks for getting back to me! I've made the changes you requested:
* b0c5dd7 has the tweaks from your code comments
* 99834d7 replaces the random values in generated data with (more compressible) ones/zeros and adds compression; it also adds a README that explains how the test data are generated.

File: tensorflow_datasets/image/nyu_depth_v2_test.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow_datasets import testing
+from tensorflow_datasets.image import nyu_depth_v2
+
+
+class NyuDepthV2Test(testing.DatasetBuilderTestCase):
+  DATASET_CLASS = nyu_depth_v2.NyuDepthV2
+  SPLITS = {"train": 2, "test": 1}
```
Comment (ChanchalKumarMaji) [inline tensorflow_datasets/image/nyu_depth_v2_test.py]: Tests Failing !!!!!!
Suggested Changes - please change to - 
`SPLITS = {"train": 2, "validation": 1}`

Comment (leonoverweel) [reply tensorflow_datasets/image/nyu_depth_v2_test.py]: Thanks, fixed in 46835075ad7dd6dd4b0478877f57b07e0e282add.

**Comment by leonoverweel** [comment]
Hey @Conchylicultor @cyfra, do you think this is ready to merge? Thanks :)

**Comment by leonoverweel** [comment]
Hey, just wanted to ping again—@Conchylicultor @cyfra, now that @ChanchalKumarMaji has approved this PR, do you think we can get it merged? Thanks :)

**Comment by ChanchalKumarMaji** [comment]
Can u please run pylint once and check if u getting 10.0 score. Thanks.

**Comment by leonoverweel** [comment]
Hey @ChanchalKumarMaji, thanks for the quick reply! I ran `./oss_scripts/lint.sh tensorflow_datasets/image` and fixed all the problems it reported in my files in 0dcf917. (It gets 9.83 but all remaining problems are in files for other datasets that I haven't touched in this PR.)

**Comment by leonoverweel** [comment]
I fixed the merge conflicts as well. @ChanchalKumarMaji et al. lmk if I need to do anything else to get this merged. Thanks!

**Comment by leonoverweel** [comment]
Hey all, anything still blocking this merge? Thanks

**Comment by leonoverweel** [comment]
@ChanchalKumarMaji, hey is there anything else I need to do to get this merged? Thanks :)

**Comment by leonoverweel** [comment]
Hey, any updates on this?

**Comment by leonoverweel** [comment]
Hey, can we get this merged or are there still blockers? Thanks :)

**Comment by leonoverweel** [comment]
@Conchylicultor thanks so much! Looks like tests are failing now that the branch has been stale for a bit, should I rebase?

**Pull request merged**